google_guest_agent_packaging_diff_tool.py

```diff
@@ -1,46 +1,111 @@
+from debian.deb822 import Deb822
+
+UBUNTU_CONTROL_FILENAME = "ubuntu_control.file"
+GOOGLE_CONTROL_FILENAME = "gce_control.file"
+
+
+@dataclass
+class ArchiveURLS:
+    ubuntu_control_url: str
+    google_upstream_control_url: str
+
+archives: dict[str, ArchiveURLS]
+
+archive_dict = {
+    "google-guest-agent": [
+        "https://git.launchpad.net/~ubuntu-core-dev/+git/google-guest-agent/plain/debian/control?h=ubuntu/master",
+        "https://raw.githubusercontent.com/GoogleCloudPlatform/guest-agent/main/packaging/debian/control"
+    ],
+    "gce-compute-image-packages": [
+        "https://git.launchpad.net/~ubuntu-core-dev/+git/gce-compute-image-packages/plain/debian/control?h=ubuntu/master",
+        "https://raw.githubusercontent.com/GoogleCloudPlatform/guest-configs/master/packaging/debian/control"
+    ],
+    "google-compute-engine-oslogin": [
+        "https://git.launchpad.net/~ubuntu-core-dev/+git/google-compute-engine-oslogin/plain/debian/control?h=ubuntu/master",
+        "https://raw.githubusercontent.com/GoogleCloudPlatform/guest-oslogin/master/packaging/debian/control"
+    ],
+    "google-osconfig-agent": [
+        "https://git.launchpad.net/~ubuntu-core-dev/+git/google-osconfig-agent/plain/debian/control?h=ubuntu/master",
+        "https://raw.githubusercontent.com/GoogleCloudPlatform/osconfig/master/packaging/debian/control"
+    ],
+} # type: archives
+
+def clean_up() -> None:
+    os.remove(UBUNTU_CONTROL_FILENAME)
+    os.remove(GOOGLE_CONTROL_FILENAME)
+
+def parse_control_file(control_file: str | os.PathLike) -> Deb822:
+    with open(control_file) as fp:
+        control_file_text = fp.read()
 
-def parse_control_file(control_file):
-    control_file_text = control_file.read()
     control_file_text_no_blank_lines = os.linesep.join([s for s in control_file_text.splitlines() if s])
     control_fileds = deb822.Deb822(control_file_text_no_blank_lines)
     return control_fileds
 
-
 @click.command()
-@click.option('--upstream-control-file', type=click.File(), required=True, help='Upstream control files to compare')
-@click.option('--ubuntu-control-file', type=click.File(), required=True, help='Ubuntu control files to compare')
-def main(upstream_control_file, ubuntu_control_file):
-    # wget --output-document=ubuntu-control "https://git.launchpad.net/ubuntu/+source/google-guest-agent/tree/debian/control?h=applied/ubuntu/noble-devel"
-    # wget --output-document=ubuntu-control "https://git.launchpad.net/ubuntu/+source/google-guest-agent/plain/debian/control?h=applied/ubuntu/noble-devel"
-    ubuntu_control_fields = parse_control_file(ubuntu_control_file)
-    upstream_control_fields = parse_control_file(upstream_control_file)
-    fields_to_compare = ['Build-Depends', 'Depends', 'Recommends', 'Suggests', 'Breaks', 'Conflicts', 'Replaces', 'Provides']
-
-    # compare the above control fields
+@click.option('--agent-package', required=True,
+              type=click.Choice(["google-guest-agent",
```

Comment:
Same opportunity here to put these literals in alpha order.

Reply:
Done, it's in the next upload :smile: